src/registrar/templates/home.html

```diff
@@ -33,26 +33,43 @@ <h2>Domains</h2>
       </thead>
       <tbody>
         {% for domain in domains %}
-        {% comment %} ticket 796
-        {% if domain.application_status == "approved" or (domain.application does not exist)  %} {% endcomment %}
         <tr>
           <th th scope="row" role="rowheader" data-label="Domain name">
             {{ domain.name }}
           </th>
           <td data-sort-value="{{ domain.created_time|date:"U" }}" data-label="Date created">{{ domain.created_time|date }}</td>
-          <td data-label="Status">{{ domain.application_status|title }}</td>
+          <td data-label="Status">
+            {% if domain.state == "unknown" or domain.state == 'dns needed'%}
+                DNS Needed
+            {% else %}
+                {{ domain.state|title }}
+            {% endif %}
+          </td>
           <td>
             <a href="{% url "domain" pk=domain.pk %}">
-              <svg
-                class="usa-icon"
-                aria-hidden="true"
-                focusable="false"
-                role="img"
-                width="24"
-              >
-                <use xlink:href="{%static 'img/sprite.svg'%}#settings"></use>
-              </svg>
-                Manage <span class="usa-sr-only">{{ domain.name }}</span>
+              {% if domain.state == "deleted" or domain.state == "on hold" %}
+                <svg
+                  class="usa-icon"
+                  aria-hidden="true"
+                  focusable="false"
+                  role="img"
+                  width="24"
+                >
+                  <use xlink:href="{%static 'img/sprite.svg'%}#visibility"></use>
+                </svg>
+                  View <span class="usa-sr-only">{{ domain.name }}</span>
+              {% else %}
+                <svg
+                  class="usa-icon"
+                  aria-hidden="true"
+                  focusable="false"
+                  role="img"
+                  width="24"
+                >
+                  <use xlink:href="{%static 'img/sprite.svg'%}#settings"></use>
+                </svg>
+                  Manage <span class="usa-sr-only">{{ domain.name }}</span>
+              {% endif %}
```

Comment:
```suggestion
                <svg
                  class="usa-icon"
                  aria-hidden="true"
                  focusable="false"
                  role="img"
                  width="24"
                >
                {% if domain.state == domain.State.DELETED or domain.state == domain.State.ON_HOLD %}
                  <use xlink:href="{%static 'img/sprite.svg'%}#visibility"></use>
                </svg>
                  View <span class="usa-sr-only">{{ domain.name }}</span>
              {% else %}
              <use xlink:href="{%static 'img/sprite.svg'%}#settings"></use>
               </svg>
               Manage <span class="usa-sr-only">{{ domain.name }}</span>
              {% endif %}
```
(Optional) Because it looks like the only thing changing between these two if statements is the href for the xlink and the text for the <a> tag, you could reduce code duplication here by squishing that if/else clause inside of it. See the semi-pseudo code above (semi because its messy but in practice its there)